builder/builder.go

```diff
@@ -62,15 +58,11 @@ type Backend interface {
 	// TODO: extract in the builder instead of passing `decompress`
 	// TODO: use containerd/fs.changestream instead as a source
 	CopyOnBuild(containerID string, destPath string, srcRoot string, srcPath string, decompress bool) error
-
-	// MountImage returns mounted path with rootfs of an image.
-	MountImage(name string) (string, func() error, error)
 }
 
-// Image represents a Docker image used by the builder.
-type Image interface {
-	ImageID() string
-	RunConfig() *container.Config
+// ImageBackend are the interface methods required from an image component
+type ImageBackend interface {
+	GetImageAndLayer(ctx context.Context, refOrID string, opts backend.GetImageAndLayerOptions) (Image, ReleaseableLayer, error)
```

Comment:
It can't be `GetImage` because it conflicts with another daemon method already called `GetImage`.

I called it Layer, since the second return is named ReleaseableLayer. Should we rename that object as well? I think they should match.

Or should I make this `GetImageAndReleasableLayer()` ?